packages/andive/src/components/filter-button.tsx

```diff
@@ -0,0 +1,65 @@
+import React, {DOMAttributes} from 'react'
+import styled, {css} from 'styled-components'
+
+import {body2Css} from './typography'
+import * as palette from '../constants/palette'
+import FilterIcon from './icons/filter'
+
+const FilterButtonRoot = styled(({active, backgroundColor, ...props}) => <button {...props} />)<{
+  active?: boolean
+  backgroundColor?: string
+}>`
+  outline: none;
+  border: none;
+  background: ${({active, backgroundColor}) =>
+    backgroundColor || (active ? palette.mediumBerryBlue : palette.mediumGrey)};
+  border-radius: 16px;
+  cursor: pointer;
+  height: 40px;
+  padding: 8px;
+  display: flex;
+  align-items: center;
+
+  ${({active}) =>
+    active &&
+    css`
+      *[data-andive-type='typography'],
+      svg {
+        color: ${palette.white};
+      }
+    `}
+
+  &:hover {
+    background-color: ${({active}) => (active ? palette.darkBerryBlue : palette.darkGrey)};
+  }
+`
+
+const FilterText = styled(({active, color, ...props}) => <div {...props} />)<{
+  active?: boolean
+  color?: string
+}>`
+  ${body2Css};
+  color: ${({active, color}) => color || (active ? palette.white : palette.secondaryText)};
+  padding: 0 8px;
+`
+
+interface FilterButtonProps {
+  className?: string
+  label?: string | JSX.Element
+  onClick?: DOMAttributes<HTMLButtonElement>['onClick']
+  color?: string
+  active?: boolean
+}
+
+export const FilterButton = ({active, color, onClick, className, label, ...buttonProps}: FilterButtonProps) => {
+  return (
+    <FilterButtonRoot onClick={onClick} className={className} active={active} {...buttonProps}>
+      <FilterIcon color={color} />
```

Comment:
I would like the icon to default to FilterIcon if undefined, but overriden if `null | React.ReactNode` (to remove it or swap it if necessary)